### ISSUE TITLE
Do we teach the danger of hard-coding?

### ISSUE THREAD

**Issue by bclaremar** [UPPMAX/programming_formalisms#143]

**Comment by richelbilderbeek** [comment]
I think the bigger question is (even) more interesting: how do we as a group decide what an individual teaches? What if that individual is wrong? I'll think about this :-)

**Comment by bclaremar** [comment]
Examples that we cover or could cover

- make analysis work for different data (or at least different files with same format)
- avoid control of program through comment/uncommenting code
    - instead use if/else and possibly arguments when running program

**Comment by richelbilderbeek** [comment]
I think this is useful and I will add it where I can if I consider this useful enough to teach within the time I have. 

Thanks!

**Comment by bclaremar** [comment]
and of course I forgot:

"using variables that can easily be defined instead of numbers at several places that may change as you decide to do updates/bug fixes."

Would you agree on just gathering a list of such "soft coding tips". May that be sufficient? Would fit where variables are covered?

Or do I have a bad understanding of what hardcoding is?
Well, Wikipedia gives a bit more narrow definition of softcoding. They say that softcoding requires external files or arguments.

**Comment by bclaremar** [comment]
https://en.wikipedia.org/wiki/Hard_coding
https://en.wikipedia.org/wiki/Hard_coding#Softcoding

**Comment by richelbilderbeek** [comment]
Yes, I think this is useful and I will add it where I can if I consider this useful enough to teach within the time I have.

I understand hard-coding. I will pick teaching good practices over teaching bad practices to be avoided (all else being equal). Still, hardcoding does deserve some attention. The question is if it will reach the priority list, not if it is on it :+1:

**Comment by richelbilderbeek** [comment]
Thanks @LinusSch, I enjoy your thoughtful post! For me to add it to my material, I do need to (1) be convinced I should teach this, and (2) have references to the literature. 

It may be overkill to already give references to the literature in an issue, yet I will ask for them by the ones that convinced me :-)

But hey, let's ask: @LinusSch: which book did you quote, or did you remember this from a now-unknown source?

**Comment by richelbilderbeek** [comment]
After another course iteration, I did not feel the need to address this. Not because it is unimportant, but because other things are even more important. 

Closing this 👍
